app/src/main/kotlin/at/bitfire/davdroid/ui/AccountsActivity.kt

```diff
@@ -6,52 +6,19 @@ package at.bitfire.davdroid.ui
 
 import android.content.Intent
 import android.os.Bundle
-import androidx.activity.compose.setContent
 import androidx.appcompat.app.AppCompatActivity
-import at.bitfire.davdroid.ui.account.AccountActivity
-import at.bitfire.davdroid.ui.intro.IntroActivity
-import at.bitfire.davdroid.ui.setup.LoginActivity
-import dagger.hilt.android.AndroidEntryPoint
-import javax.inject.Inject
 
-
-@AndroidEntryPoint
+/**
+ * Legacy activity, previously used to host [AccountsScreen].
+ * Needed for backwards compatibility with external apps / shortcuts that call the activity directly.
+ */
 class AccountsActivity: AppCompatActivity() {
 
-    @Inject
-    lateinit var accountsDrawerHandler: AccountsDrawerHandler
-
-    private val introActivityLauncher = registerForActivityResult(IntroActivity.Contract) { cancelled ->
-        if (cancelled)
-            finish()
-    }
-
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        // handle "Sync all" intent from launcher shortcut
-        val syncAccounts = intent.action == Intent.ACTION_SYNC
-
-        setContent {
-            AccountsScreen(
-                initialSyncAccounts = syncAccounts,
-                onShowAppIntro = {
-                    introActivityLauncher.launch(null)
-                },
-                accountsDrawerHandler = accountsDrawerHandler,
-                onAddAccount = {
-                    startActivity(Intent(this, LoginActivity::class.java))
-                },
-                onShowAccount = { account ->
-                    val intent = Intent(this, AccountActivity::class.java)
-                    intent.putExtra(AccountActivity.EXTRA_ACCOUNT, account)
-                    startActivity(intent)
-                },
-                onManagePermissions = {
-                    startActivity(Intent(this, PermissionsActivity::class.java))
-                }
-            )
+        MainActivity.legacyRedirect(this) {
+            putExtra(MainActivity.EXTRA_SYNC_ACCOUNTS, action == Intent.ACTION_SYNC)
```

Comment:
I wonder whether there's a canonical way of passing the route to the activity. So instead of passing the data to the `MainActivity` (which then creates the correct destination), can we directly pass the compose destination to the `MainActivity`?

The same question occurs when we use a `PendingIntent` for a notification – in this case the Intent must contain the actual destination. As I [understand the docs, (non-public) deep links](https://developer.android.com/develop/ui/compose/navigation#deeplinks) can be used for that.

However I must say that I don't know what's the correct way. We should find out, we have to know it at latest when we migrate notifications.

Reply:
I've now migrated to use deep links, but I'm not sure still if that is the best method. Can you take a look?

Reply:
I'll dig into it again (will take some time). Maybe we can have a talk together at some time @ArnyminerZ @sunkup because I think it will be important for all of us. Ideally with a short summary on what we need to know to migrate everything to Compose.